kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -443,7 +445,7 @@ ensureSimplifiedResult
     -> OrPattern variable
     -> simplifier (OrPattern variable)
 ensureSimplifiedResult repr termLike results
-  | isSimplified results = pure results
+  | hasSimplifiedChildren results = pure results
```

Comment:
We should revert this change, because the `TermLike` simplifier should actually return a fully-simplified condition.